pkg/agent/cniserver/interface_configuration_linux.go

```diff
@@ -44,6 +51,115 @@ func newInterfaceConfigurator(ovsDatapathType string) (*ifConfigurator, error) {
 	return &ifConfigurator{ovsDatapathType: ovsDatapathType}, nil
 }
 
+func renameLink(curName, newName string) error {
+	link, err := netlink.LinkByName(curName)
+	if err != nil {
+		return err
+	}
+	if err := netlink.LinkSetDown(link); err != nil {
+		return err
+	}
+	if err := netlink.LinkSetName(link, newName); err != nil {
+		return err
+	}
+	if err := netlink.LinkSetUp(link); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func moveIfToNetns(ifname string, netns ns.NetNS) error {
+	vfDev, err := netlink.LinkByName(ifname)
+	if err != nil {
+		return fmt.Errorf("failed to lookup vf device %v: %q", ifname, err)
+	}
+	// move VF device to ns
+	if err = netlink.LinkSetNsFd(vfDev, int(netns.Fd())); err != nil {
+		return fmt.Errorf("failed to move device %+v to netns: %q", ifname, err)
+	}
+
+	return nil
+}
+
+// setupSriovInterfaces move VF to the container namesapce
+func (ic *ifConfigurator) setupSriovInterfaces(
```

Comment:
Per our discussion, I assume you will check hw-offload configured here (I prefer to check it at agent startup, e.g. in setupOVSBridge and save the state to NodeConfig)?

Reply:
but how will I know in agent startup that it is running in offload mode? which means it configure with multus to pass the VF DeviceID?

Reply:
Sorry, maybe I did not express it clearly. I just meant at agent startup, agent could detect if hw-offload is configured for the bridge and save the state (e.g. to NodeConfig).  Then when receiving a CNI ADD with DeviceID, agent can check the save state.

Reply:
I believe Jianjun means by reading `other-config:hw-offload` from OVSDB during Agent initialization. I believe the `GetOVSOtherConfig` method can be used (https://github.com/vmware-tanzu/antrea/blob/v0.8.2/pkg/ovs/ovsconfig/interfaces.go#L48).

Reply:
Thanks for addressing this comment.